demo.yml

```diff
@@ -33,30 +36,46 @@
         port_range_min: 22
         port_range_max: 22
         remote_ip_prefix: "{{ ansible_default_ipv4.address }}/32"
+    
+    - name: Creates a key pair with the running users public key
+      os_keypair:
+          state: present
+          name: "{{ demo_key }}"
+          public_key_file: "{{ path_to_key }}"
 
     - name: Create a virtual machine
       register: result
       os_server:
+        state: present
         name: pouta-demo
```

Comment:
Use a variable instead, again I find it more user friendly for the non ansible people.